src/net/sourceforge/kolmafia/textui/command/AbsorbCommand.java

```diff
@@ -68,6 +68,19 @@ public void run(final String cmd, String parameters) {
       PreferenceListenerRegistry.firePreferenceChanged("(hats)");
     }
 
-    KoLmafia.updateDisplay("Absorbed " + (count > 1 ? count + " " : "") + match.getPluralName());
+    int failedAbsorbs = count - (KoLCharacter.getAbsorbs() - previousAbsorbs);
+    count -= failedAbsorbs;
+
+    if (count > 0) {
+      KoLmafia.updateDisplay("Absorbed " + (count > 1 ? count + " " : "") + match.getPluralName());
```

Comment:
the way `match.getPluralName` works is that if `match`'s count is 1, it's the single, otherwise the plural. Now that you're adjusting `count` this won't work correctly any more (same for `failedAbsorbs`).

Instead, use `ItemDatabase.getPluralName` and `match.getName`, and swap them in as appropriate.

Reply:
But an int isn't passed by reference. It hasn't modified `match` because `count` is an `int`, not something like an `AtomicInteger` or something where we pass the reference around.

Reply:
Oh wait, sorry. I just realized